elastic/integrations
# Adapt definitions of nested subfields to current Fleet implementation

## Proposed commit message

Adapt definitions of subfields of nested objects so they are actually installed by Fleet. New mappings are otherwise equivalent.

These mappings were correctly defined as expected by the spec, but Fleet was only installing empty nested objects. To workaround that, subfields can be moved to have their own definitions.

Issue in Fleet is fixed in https://github.com/elastic/kibana/pull/191730, but we can apply this workaround for older versions of the stack. 

## Checklist


## How to test this PR locally

Install one of these packages, and check that the index templates include the mappings for the fields of the nested objects.

## Related issues


- Closes https://github.com/elastic/package-spec/issues/784
- Related to https://github.com/elastic/kibana/pull/191730
- Originally part of to https://github.com/elastic/integrations/pull/10919
